File: common/types/note/NoteInterval.java
package com.ariasproj.common.types.note;

import com.ariasproj.common.exceptions.InvalidIntervalException;

public class NoteInterval {

	private byte interval;
	public NoteInterval(){
		this.interval = 0;
	}
	public NoteInterval(byte interval) throws InvalidIntervalException{
		if (interval < 1 || interval > 8)
			throw new InvalidIntervalException();
		
		this.interval = interval;
	}

	public byte getInterval(){
		return this.interval;
	}
}

File: source/common/types/note/noteinterval.cpp


#include "noteinterval.h"
#include "invalidintervalexception.h"
using namespace arias::common::types::note;
using namespace arias::common::exceptions;


NoteInterval::NoteInterval()
{
	this->interval = 0;
}

NoteInterval::NoteInterval(uint8_t interval) 
{
	if (interval < 1 || interval > 8)
		throw InvalidIntervalException();
		
	this->interval = interval;
}

uint8_t NoteInterval::getInterval()
{
	return this->interval;
}

